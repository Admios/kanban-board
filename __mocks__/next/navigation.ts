const push = jest.fn();
export const useRouter = () => ({ push });
export const redirect = jest.fn();
export const navigation = { push, redirect };
