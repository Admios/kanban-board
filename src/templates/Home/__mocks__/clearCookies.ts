export const clearCookies = jest.fn().mockResolvedValue(undefined);
