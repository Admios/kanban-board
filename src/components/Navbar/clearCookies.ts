"use server";

import { cookies } from "next/headers";

export async function clearCookies() {
  cookies()
    .getAll()
    .forEach((cookie) => {
      cookies().delete(cookie.name);
    });
}
