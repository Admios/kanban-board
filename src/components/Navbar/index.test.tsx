import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { useRouter } from "next/navigation";
import { Navbar } from ".";
import { clearCookies } from "./clearCookies";

jest.mock("./clearCookies.ts");

it("should show login when there is no user", async () => {
  const { push } = (useRouter as jest.Mock)();
  const { container } = render(<Navbar navbarItems={[]} />);
  expect(container).toMatchSnapshot("Default Header");

  // Click the button
  const button = await screen.findByText("Login");

  expect(button).toBeInTheDocument();
});

it("should launch logout when button is pressed", async () => {
  const { container } = render(
    <Navbar user={{ email: "test" }} navbarItems={[]} />,
  );
  expect(container).toMatchSnapshot("Logged in Header");

  // Click the button
  const button = await screen.findByText("Logout");
  expect(button).toBeInTheDocument();

  await userEvent.click(button);
  expect(clearCookies).toHaveBeenCalled();
});
