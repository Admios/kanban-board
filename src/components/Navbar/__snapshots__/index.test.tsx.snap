// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should launch logout when button is pressed: Logged in Header 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item"
        href="/"
      >
        <img
          alt="Logo"
          data-nimg="1"
          decoding="async"
          height="28"
          loading="lazy"
          src="/admios-logo.svg"
          style="color: transparent;"
          width="112"
        />
      </a>
      <a
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger"
        data-target="main-navbar-inner"
        role="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      class="navbar-menu"
      id="main-navbar-inner"
    >
      <div
        class="navbar-start"
      >
        <a
          class="navbar-item"
          href="/"
        >
          My Boards
        </a>
      </div>
      <div
        class="navbar-end"
      >
        <div
          class="navbar-item"
        >
          Username: 
          test
        </div>
        <a
          class="navbar-item"
        >
          Logout
        </a>
      </div>
    </div>
  </nav>
</div>
`;

exports[`should show login when there is no user: Default Header 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item"
        href="/"
      >
        <img
          alt="Logo"
          data-nimg="1"
          decoding="async"
          height="28"
          loading="lazy"
          src="/admios-logo.svg"
          style="color: transparent;"
          width="112"
        />
      </a>
      <a
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger"
        data-target="main-navbar-inner"
        role="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      class="navbar-menu"
      id="main-navbar-inner"
    >
      <div
        class="navbar-start"
      >
        <a
          class="navbar-item"
          href="/"
        >
          My Boards
        </a>
      </div>
      <div
        class="navbar-end"
      >
        <div
          class="navbar-item"
        >
          Username: 
          guest
        </div>
        <a
          class="navbar-item"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </nav>
</div>
`;
